### ISSUE TITLE
[Reader] seek to batch messageId went wrong in 2.5.0

### ISSUE THREAD

**Issue by yjshen** [apache/pulsar#6344]
Starting the reader with a MessageId inside a batch:  (11:0:3:151) with :
```
client.newReader()
         .topic(topic)
         .startMessageId(startMessageId)
         .startMessageIdInclusive()
         .loadConf(readerConf)
         .create();
```
the first message I read with reader.readNext() happens to be the one with messageId (11:0:3:0).

This is 2.5.0 related since my library relies on 2.4.2 works correctly.

**Comment by lhotari** [comment]
I wonder if #6333 is related.

**Comment by devinbost** [comment]
@yjshen Do you think this one might also be related? 
If we can already reproduce this in a test, then we have a clear path to fixing it.

**Comment by devinbost** [comment]
@yjshen Do you think this one might also be related? 
If we can already reproduce this in a test, then we have a clear path to fixing it. 
https://github.com/apache/pulsar/issues/6224

**Comment by yjshen** [comment]
Hi @lhotari @devinbost , I found this bug is brought in by #5622 , and I fixes it in #6345 . I'll look into #6333 and #6224 now.

**Comment by yjshen** [comment]
#6224 is a bug but not related to this issue. @devinbost will fix it in his PR.
